### PR TITLE
[Kotlin] - Convert files from groovy to Kotlin

### LINKED ISSUE #150: Language clean up + Add Kotlin support
Can we look into cleaning up the code?
 - Adding documentation
 - Removing `Java` code from the `Groovy` `src/groovy` folder - https://github.com/KeepSafe/dexcount-gradle-plugin/pull/152

Also, have we considered looking into kotlin?

### PR DESCRIPTION
**Pull request by jaredsburrows**
@benjamin-bader 
Slowly converting files from Groovy to Kotlin
Closes #157 
Closes #150

### CODE REVIEW

**Comment by jaredsburrows** [comment]
@benjamin-bader I have converted about half of the code base to Kotlin. Let me know what you think!

File: src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2015-2016 KeepSafe Software
```
Comment (benjamin-bader) [inline src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt]: Why remove the date range?

File: src/main/kotlin/com/getkeepsafe/dexcount/PrintOptions.kt [hunk 2]
```diff
@@ -16,14 +16,14 @@
...
-    def includeClasses = false
-    def includeClassCount = false
-    def includeMethodCount = true
-    def includeFieldCount = false
-    def includeTotalMethodCount = false
-    def teamCityIntegration = false
-    def printHeader = false
-    def orderByMethodCount = false
-    def int maxTreeDepth = Integer.MAX_VALUE
+class PrintOptions {
```
Comment (benjamin-bader) [inline src/main/kotlin/com/getkeepsafe/dexcount/PrintOptions.kt]: This should probably be `internal` to approximate package-privacy.

File: src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt (hunk 1 above)
Comment (jaredsburrows) [inline src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt]: I just used the same copyright for all the files, the 2016 version. Did you want me to add it back?

File: src/main/kotlin/com/getkeepsafe/dexcount/PrintOptions.kt (hunk 2 above)
Comment (jaredsburrows) [inline src/main/kotlin/com/getkeepsafe/dexcount/PrintOptions.kt]: Ok.

File: src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt (hunk 1 above)
Comment (benjamin-bader) [inline src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt]: Yeah, I'd prefer if we kept it intact.  It seems likely to me that these Kotlin files would be construed as derivative works, so best to keep it safe (no pun intended).

**Comment by benjamin-bader** [comment]
I like this approach!  Let's get it all transliterated from Groovy -> Kotlin first, then worry about idiomatic Kotlin later.

**Comment by jaredsburrows** [comment]
@benjamin-bader Translate it all to Kotlin in a single PR?

File: src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt (hunk 1 above)
Comment (jaredsburrows) [inline src/main/kotlin/com/getkeepsafe/dexcount/OutputFormat.kt]: Ok.

**Comment by benjamin-bader** [comment]
No, sorry - I just meant that I like the approach taken in this PR.  We can land this one, and do more later.

**Comment by jaredsburrows** [comment]
@benjamin-bader Thanks! I have mad the fixes you have asked.

**Pull request merged**